github_stargazers/github.py

```diff
@@ -17,6 +17,12 @@ def __init__(self) -> None:
         super().__init__("Too many requests.")
 
 
+class UrlNotFoundError(Exception):
+    def __init__(self, repository) -> None:
+        repository_display = repository or ""
+        super().__init__(f"Resource not Found. Check that the repository {repository_display} is correct.")
```

Comment:
Lets drop the extra-variable and put the content of it there in the f-string. `{repository or ""}`